File: app/src/main/java/com/sharpdroid/registroelettronico/Views/SubjectDetails/OverallView.java
package com.sharpdroid.registroelettronico.Views.SubjectDetails;

import android.content.Context;
import android.support.v7.widget.CardView;
import android.util.AttributeSet;
import android.widget.TextView;

import com.sharpdroid.registroelettronico.R;

import java.util.Locale;

import butterknife.BindView;
import butterknife.ButterKnife;
import io.reactivex.annotations.Nullable;

public class OverallView extends CardView {
    Context mContext;
    AttributeSet mAttributeSet;
    int defStyleAttr;

    @BindView(R.id.scritto)
    TextView mScritto;
    @BindView(R.id.orale)
    TextView mOrale;
    @BindView(R.id.pratico)
    TextView mPratico;
    @BindView(R.id.text1)
    TextView text1;
    @BindView(R.id.text2)
    TextView text2;
    @BindView(R.id.text3)
    TextView text3;
    @BindView(R.id.header)
    TextView header;

    public OverallView(Context context) {
        super(context);
        mContext = context;
        init();
    }

    public OverallView(Context context, AttributeSet attrs) {
        super(context, attrs);
        mContext = context;
        mAttributeSet = attrs;
        init();
    }

    public OverallView(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        mContext = context;
        mAttributeSet = attrs;
        this.defStyleAttr = defStyleAttr;
        init();
    }

    void init() {
        inflate(mContext, R.layout.view_overall_subject, this);
        ButterKnife.bind(this);
    }

    public void setScritto(@Nullable Float scritto) {
        if (scritto == null) mScritto.setText("-");
        else mScritto.setText(String.format(Locale.getDefault(), "%.2f", scritto));
    }

    public void setOrale(@Nullable Float orale) {
        if (orale == null) mOrale.setText("-");
        else mOrale.setText(String.format(Locale.getDefault(), "%.2f", orale));
    }

    public void setPratico(@Nullable Float pratico) {
        if (pratico == null) mPratico.setText("-");
        else mPratico.setText(String.format(Locale.getDefault(), "%.2f", pratico));
    }
}

File: app/src/main/java/com/sharpdroid/registroelettronico/views/subjectDetails/OverallView.kt
package com.sharpdroid.registroelettronico.views.subjectDetails

import android.content.Context
import android.support.v7.widget.CardView
import android.util.AttributeSet
import android.view.View
import com.sharpdroid.registroelettronico.R
import kotlinx.android.synthetic.main.view_overall_subject.view.*
import java.util.*

class OverallView : CardView {
    internal var mContext: Context

    constructor(context: Context) : super(context) {
        mContext = context
        init()
    }

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs) {
        mContext = context
        init()
    }

    constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int) : super(context, attrs, defStyleAttr) {
        mContext = context
        init()
    }

    internal fun init() {
        View.inflate(mContext, R.layout.view_overall_subject, this)
    }

    fun setScritto(Scritto: Float?) {
        if (Scritto == null)
            scritto.text = "-"
        else
            scritto.text = String.format(Locale.getDefault(), "%.2f", Scritto)
    }

    fun setOrale(Orale: Float?) {
        if (Orale == null)
            orale.text = "-"
        else
            orale.text = String.format(Locale.getDefault(), "%.2f", Orale)
    }

    fun setPratico(Pratico: Float?) {
        if (Pratico == null)
            pratico.text = "-"
        else
            pratico.text = String.format(Locale.getDefault(), "%.2f", Pratico)
    }
}
